lib/user.rb

```diff
@@ -0,0 +1,24 @@
+class User
+
+attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+    @jokes
+  end
+
+  def tell(recipient, joke)
+    recipient.jokes << joke
+  end
```

Comment:
generally, it's bad practice for an Object to directly write data into another Object; is there some other way that we could teach Ali this joke?